XOGame/BoardGame.cs

```diff
@@ -0,0 +1,179 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace XOGame
+{
+	public class BoardGame : IBoardGame
+	{
+		public int OScore { get; set; }
+		public int XScore { get; set; }
+		public string[] Slot { get; set; }
+		public string CurrentTurn { get; set; }
+
+		public BoardGame()
+		{
+			Slot = new string[9];
+		}
+
+		#region Method Section
+
+		#region Public Method
+		public string GetWinner()
+		{
+			return CheckWin();
+		}
+
+		public bool TakeSlot(bool isX, int row, int column)
+		{
+			bool currentTurn = false;
+			int index;
+
+			bool isProcess = VerifyInput(row, column, out index) ? true : false;
+
+			if (isProcess)
+			{
+				currentTurn = CheckCurrentTurn();
+				if (CheckSlotCanPlace(index)) // ถ้าเป็น null และเป็นตาที่ถูกต้อง ให้ลงได้
+				{
+					Slot[index] = CurrentTurn = currentTurn ? "X" : "O";
+					return true;
+				}
+				else
+				{
+					return false;
+				}
+			}
+			else
+			{
+				return false;
+			}
+		}
+
+		public void Board()
```

Comment:
ให้แยกการแสดงผลออกจากการคำนวณครับ เพราะไม่อย่างนั้นเวลาที่เราจะเอาเกมนี้ไปเล่นบน เว็บ, มือถือ หรืออะไรก็ตามแต่ เราจะต้องมาแก้โค้ดการคำนวณทุกครั้ง ซึ่งผิดหลักของ Separation of concern ครับ